d2format/escape.go

```diff
@@ -47,10 +47,6 @@ func escapeUnquotedValue(s string, inKey bool) string {
 		return `""`
 	}
 
-	if strings.EqualFold(s, "null") {
-		return "\\null"
```

Comment:
The right fix should be changing this to `"\\\\null"`

Reply:
The idea is that this null string should be parsed not as the value null but as the string null.

You could also just make it \`'null'\` to make it more clear.